app/controllers/preferences_controller.rb

```diff
@@ -1,15 +1,52 @@
 class PreferencesController < ApplicationController
+  before_action :set_preference, only: %i[edit update destroy show]
+
   def index
     @preferences = current_user.preferences
     @pagy, @records = pagy(@preferences)
   end
-  
+
+  def show
+  end
+
   def new
     @preference = Preference.new
   end
 
+  def edit
+  end
+
   def create
-    preference_params = params.require(:preference).permit(:name, :description, :restriction)
     @preference = current_user.preferences.build(preference_params)
+
+    if @preference.save
+      redirect_to preferences_path, notice: 'Preference was created successfully'
```

Comment:
What if we move this kind of texts in notices and alerts to `locales`? See https://guides.rubyonrails.org/i18n.html